broadcastproto/transport/udp.go

```
@@ -46,37 +48,193 @@ func (u *udpProto) PackagingMode() TsPackagingMode {
+	out := &LiveUrl{}
+
+	u, err := url.Parse(urlStr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid URL: %w", err)
+	}
+	out.Scheme = u.Scheme
+
+	host, portStr, err := net.SplitHostPort(u.Host)
+	if err != nil {
+		return nil, fmt.Errorf("invalid host:port in URL (%s): %w", u.Host, err)
+	}
+	if strings.HasPrefix(host, "@") {
+		return nil, fmt.Errorf("invalid host '%s': '@' prefix is not supported", host)
+	}
+	out.Host = host
+
+	port, err := strconv.Atoi(portStr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid port %s: %w", portStr, err)
+	}
+	out.Port = port
+
+	ip := net.ParseIP(host)
+	if ip == nil {
+		addrs, lookupErr := net.LookupIP(host)
+		if lookupErr != nil {
+			return nil, fmt.Errorf("unable to resolve host '%s': %w", host, lookupErr)
+		}
+		if len(addrs) == 0 {
+			return nil, fmt.Errorf("unable to resolve host '%s': no addresses returned", host)
+		}
+		for _, candidate := range addrs {
+			if candidate.To4() != nil {
+				ip = candidate
+				break
+			}
+		}
+		if ip == nil {
+			ip = addrs[0]
+		}
+	}
+	if ip == nil {
+		return nil, fmt.Errorf("unable to determine IP for host '%s'", host)
+	}
+	out.Addr = &net.UDPAddr{IP: ip, Port: out.Port}
+	if ip.IsMulticast() {
+		out.Multicast = true
+		out.Group = &net.UDPAddr{IP: ip, Port: out.Port}
+	}
+
+	// Parse query parameters
+	q := u.Query()
+
+	if la := q.Get("localaddr"); la != "" {
+		laIp := net.ParseIP(la)
+		if laIp == nil {
+			return nil, fmt.Errorf("localaddr is not a valid IP: %s", la)
+		}
+		out.LocalAddr = laIp
+	}
+
+	if srcs := q.Get("sources"); srcs != "" {
+		for _, s := range strings.Split(srcs, ",") {
+			s = strings.TrimSpace(s)
+			if s == "" {
+				continue
+			}
+			ip := net.ParseIP(s)
+			if ip == nil {
+				return nil, fmt.Errorf("invalid source IP: %s", s)
+			}
+			out.Sources = append(out.Sources, ip)
+		}
+	}
+
+	if q.Get("reuse") == "1" {
+		out.Reuse = true
+	}
+
+	return out, nil
+}
+
+func interfaceByIP(ip net.IP) (*net.Interface, error) {
```

The function silently continues on iface.Addrs() error. Consider logging or accumulating errors.